cmd/sealer/cmd/cluster/run-app.go

```diff
@@ -0,0 +1,144 @@
+// Copyright © 2021 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/sealerio/sealer/cmd/sealer/cmd/types"
+	"github.com/sealerio/sealer/common"
+	clusterruntime "github.com/sealerio/sealer/pkg/cluster-runtime"
+	"github.com/sealerio/sealer/pkg/clusterfile"
+	v12 "github.com/sealerio/sealer/pkg/define/image/v1"
+	"github.com/sealerio/sealer/pkg/define/options"
+	"github.com/sealerio/sealer/pkg/imagedistributor"
+	"github.com/sealerio/sealer/pkg/imageengine"
+	"github.com/sealerio/sealer/pkg/infradriver"
+	"github.com/sealerio/sealer/pkg/registry"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var appFlags *types.APPFlags
+var exampleForRunAppCmd = ``
+
+func NewRunAPPCmd() *cobra.Command {
+	runCmd := &cobra.Command{
+		Use:     "run-app",
+		Short:   "start to run an application cluster image",
+		Long:    `sealer run-app localhost/nginx:v1`,
+		Example: exampleForRunAppCmd,
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var (
+				cf              clusterfile.Interface
+				clusterFileData []byte
+				err             error
+			)
+
+			//todo grab more cluster info from api server
+			clusterFileData, err = ioutil.ReadFile(common.GetDefaultClusterfile())
+			if err != nil {
+				return err
+			}
+
+			cf, err = clusterfile.NewClusterFile(clusterFileData)
+			if err != nil {
+				return err
+			}
+
+			cluster := cf.GetCluster()
+			infraDriver, err := infradriver.NewInfraDriver(&cluster)
+			if err != nil {
+				return err
+			}
+
+			var (
+				clusterHosts = infraDriver.GetHostIPList()
+				launchCmds   = appFlags.LaunchCmds
+				appImageName = args[0]
+			)
+
+			imageEngine, err := imageengine.NewImageEngine(options.EngineGlobalConfigurations{})
+			if err != nil {
+				return err
+			}
+
+			extension, err := imageEngine.GetSealerImageExtension(&options.GetImageAnnoOptions{ImageNameOrID: appImageName})
+			if err != nil {
+				return fmt.Errorf("failed to get cluster image extension: %s", err)
+			}
+
+			if extension.Type != v12.AppInstaller {
```

Comment:
extension Type should be named as AppImage, rather than AppInstaller

Reply:
`Type` was already defined at SealerImageExtension. KubeInstaller = "kube-installer" , AppInstaller  = "app-installer"